packages/path_provider/path_provider_android/lib/path_provider_android.dart

```diff
@@ -81,7 +81,13 @@ class PathProviderAndroid extends PathProviderPlatform {
   }
 
   @override
-  Future<String?> getDownloadsPath() {
-    throw UnsupportedError('getDownloadsPath is not supported on Android');
+  Future<String?> getDownloadsPath() async {
+    final List<String>? paths =
+        await getExternalStoragePaths(type: StorageDirectory.downloads);
+    if (paths != null && paths.isNotEmpty) {
+      return paths.first;
+    } else {
+      throw UnsupportedError('getDownloadsPath is not supported on Android');
```

Comment:
`UnsupportedError` is incorrect here; the function is supported. This would kill someone's app (because it's an Error, which isn't meant to be caught) if the underlying API call failed, which is not a programming error by the caller.

Why isn't this just returning `null`?

Reply:
in hindsight, i was thinking to notify the user that downloads path is not found. but the function returns a nullable string and the main function getDownloadsDirectory also returns a nullable directory, hence returning null should be correct.